pyproject.toml

```diff
@@ -88,39 +88,57 @@ Documentation = "https://papis.readthedocs.io"
 Repository = "https://github.com/papis/papis"
 
 [project.optional-dependencies]
-develop = [
+# Dev environment -- install if you're working on papis
+develop = ["papis[lint,typing,test]"]
+# Style enforcement
+lint = [
     "flake8",
     "flake8-bugbear",
     "flake8-quotes",
     "Flake8-pyproject",
-    "mypy>=0.7",
     "pep8-naming",
     "pylint",
+]
+# Test suite
+test = [
     "pytest",
     "pytest-cov",
+]
+# Type-checking
+typing = [
+    "mypy>=0.7",
     "types-beautifulsoup4",
     "types-Pygments",
     "types-docutils",
-    "types-python-slugify",
     "types-PyYAML",
     "types-requests",
 ]
+# To generate documentation
 docs = [
     "pytest",
     "sphinx>=4",
     "sphinx_click",
     "sphinx_design",
     "sphinx_rtd_theme>=1",
 ]
-lsp = [
-    "python-lsp-server",
-]
+# Libraries that papis knows how to interact with if installed
 optional = [
     "Jinja2>=3.0.0",
     "Whoosh>=2.7.4",
     "chardet>=3.0.2",
     "markdownify>=0.11.6",
 ]
+# Plugins using papis.sphinx_ext need these
+external_plugin = [
+    "docutils",
+    "sphinx",
+    "sphinx-click",
```

Comment:
I think `docutils` is a dependency of sphinx already, so probably not needed explicitly here?

Reply:
It is an explicit dependency of `sphinx_ext` -- it's a better idea to be explicit here, so we continue working if eg one day Sphinx decides to drop the `docutils` dependency (unlikely, but you never know)